torchaudio/datasets/iemocap.py

```diff
@@ -0,0 +1,140 @@
+import os
+import re
+from collections import defaultdict
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import _load_waveform
+
+
+SAMPLE_RATE = 16000
+
+
+def _get_wavs_paths(data_dir):
+    wav_dir = os.path.join(data_dir, "sentences/wav")
+    wav_paths = sorted(str(p) for p in Path(wav_dir).glob("*/*.wav"))
+    relative_paths = []
+    for wav_path in wav_paths:
+        start = wav_path.find("Session")
+        wav_path = wav_path[start:]
+        relative_paths.append(wav_path)
+    return relative_paths
+
+
+class IEMOCAP(Dataset):
+    """*IEMOCAP* :cite:`iemocap` dataset.
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found
+        sessions (Optional[List[int]], optional): List of sessions (1-5) to use. If ``None``, loads from all sessions.
+    """
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        sessions: Optional[str] = None,
+    ):
+        root = os.fspath(root)
+        self._path = os.path.join(root, "IEMOCAP")
+
+        if not os.path.isdir(self._path):
+            raise RuntimeError("Dataset not found.")
+
+        all_data = []
+        self.data = []
+        self.mapping = defaultdict()
+        if sessions is None:
+            sessions = [1, 2, 3, 4, 5]
+
+        for session in sessions:
+            session_name = f"Session{session}"
+            session_dir = os.path.join(self._path, session_name)
+
+            # get wav paths
+            wav_paths = _get_wavs_paths(session_dir)
+            for wav_path in wav_paths:
+                wav_stem = str(Path(wav_path).stem)
+                all_data.append(wav_stem)
+
+            # add labels
+            label_dir = os.path.join(session_dir, "dialog/EmoEvaluation")
+            label_paths = Path(label_dir).glob("*.txt")
+
+            for label_path in label_paths:
+                with open(label_path, "r") as f:
+                    for line in f:
+                        if line[0] != "[":
```

Comment:
Could we use `startwith` just for peace of mind? If the file ends with a line without break, then this part could fail. (though I know that's not expected)